app/Models/Foundation/Summit/Registration/PromoCodes/SummitRegistrationPromoCode.php

```diff
@@ -308,10 +317,23 @@ public function hasQuantityAvailable(): bool
      * @param string $email
      * @param null|string $company
      * @return bool
-     * @throw ValidationException
+     * @throws ValidationException
      */
-    public function checkSubject(string $email, ?string $company): bool
-    {
+    public function checkSubject(string $email, ?string $company): bool {
```

Comment:
maitain original implementation of checkSubject and create a new method ( ie validate ?) that performs the 3 original validations from summit order service

Reply:
done